packages/rollup-full-node/src/app/utils/local-l1-node.ts

```diff
@@ -5,31 +5,42 @@ import { Contract, ethers, providers, Wallet } from 'ethers'
 import { createMockProvider, deployContract, getWallets } from 'ethereum-waffle'
 
 /* Internal Imports */
-import { DEFAULT_ETHNODE_GAS_LIMIT } from '../index'
+import { DEFAULT_ETHNODE_GAS_LIMIT, Environment } from '../index'
+import { L1NodeContext } from '../../types'
 
-const finalityDelayInBlocks: string =
-  process.env.FINALITY_DELAY_IN_BLOCKS || '0'
-const l1NodeLevelDBPath: string = process.env.L1_NODE_LEVELDB_PATH
+/**
+ * Starts a local node on the provided port, using the provided mnemonic to
+ * deploy the necessary contracts for bootstrapping.
+ *
+ * @param mnemonic
+ * @param port
+ */
 export const startLocalL1Node = async (
```

Comment:
Hmm the terminology here is a little confusing. Am I getting it right that "LocalL1Node" is what we send "InternalTransactions" to? It's just really confusing and I want to make sure we have all of the components mapped out with reasonable terminology. 

I just made a diagram which illustrates all of the components.

![IMG_C31CC24C2F6A-1](https://user-images.githubusercontent.com/706123/77577037-84d1e580-6eac-11ea-95f8-945c197e73ff.jpeg)

So to be clear, `L1 Node` would be just a normal Geth instance that's syncing mainnet Ethereum. Then an `L2 Node` or maybe more specifically `Rollup Node` would be the thing that runs an `Internal` instance of Geth / EVM, which itself either emulates an OVM deployed to it (ExecutionManger, etc).

Does that match up with the terminology being used here?

Reply:
Yep, that's it! The line in question is to specifically start an L1 node locally to fully exercise the end-to-end functionality of L2 <--> L1 messaging. This is completely different than the L2 Node. I went with L1 and L2 because nothing is actually "rolled up" in the L2 node, and instead, the rollup state sits in the L1 node. Wanted to avoid that confusion. Does L1 and L2 sound good? Happy to rename it to rollup if you think it makes more sense.

Reply:
Ah great! So I think after staring at this for a while I see where my confusion was. The fact that inside the `LocalL1Node` we have a `sequencerWallet` was throwing me off. But it’s just because in this circumstance the L1 that we are running is a Local testnet run by the sequencer. 

These L1 & L2 terms are a little confusing! Especially when you see L1ToL2 right next to L2ToL1 — my dyslexia goes crazy! That said, it’s not clear to me that `EthNode` and `RollupNode` (for example) are much better 😕 Especially because both are in some sense “Eth” nodes.

I realize this could require some refactoring/renaming (and so I’m happy to not do it) but I wonder if instead of `L1ToL2`/`L2ToL1`, we could use the terms `L2InboundMessages` and `L2OutboundMessages` or something? Just spitballing and curious what you / @ben-chain think.

It’s a nitpick though & might be specific to my difficulty reading 😂

Reply:
Oh nvm I take it back. I’m now thinking about `l2ToL1MessageReceiver` and calling it `l2OutboundMessageReceiver` and realize that these variable names are going to be confusing basically no matter how we slice it.

Unless y'all absolutely love my naming change (which might have been closer to what Ben did originally?) doesn’t feel worth the switching cost.